sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -74,27 +85,105 @@ private static Kin getInstance() {
 		return instance;
 	}
 
-	public static void start(@NonNull Context appContext, @NonNull WhitelistData whitelistData,
-		@NonNull KinEnvironment environment)
-		throws ClientException, BlockchainException {
+	public synchronized static void initialize(Context appContext) throws ClientException {
 		if (isInstanceNull()) {
-			SignInData signInData = getWhiteListSignInData(whitelistData);
-			init(appContext, signInData, environment);
+			instance = getInstance();
+			// use application context to avoid leaks.
+			appContext = appContext.getApplicationContext();
+
+			//Load data from manifest, can throw ClientException if no data available.
+			loadDefaultsFromMetadata(appContext);
+
+			//Set Environment
+			ConfigurationImpl.init(ConfigurationLocal.getInstance(appContext));
+			ConfigurationImpl.getInstance().setEnvironment(environmentName);
+			KinEnvironment kinEnvironment = ConfigurationImpl.getInstance().getEnvironment();
+			eventLogger = EventLoggerImpl.getInstance();
+			final String networkUrl = kinEnvironment.getBlockchainNetworkUrl();
+			final String networkId = kinEnvironment.getBlockchainPassphrase();
+			final String issuer = kinEnvironment.getIssuer();
+
+			KinClient kinClient = new KinClient(appContext, new ServiceProvider(networkUrl, networkId) {
+				@Override
+				protected String getIssuerAccountId() {
+					return issuer;
+				}
+			}, KIN_ECOSYSTEM_STORE_PREFIX_KEY);
+			BlockchainSourceImpl.init(eventLogger, kinClient, BlockchainSourceLocal.getInstance(appContext));
+
+			AuthRepository
+				.init(AuthLocalData.getInstance(appContext), AuthRemoteData.getInstance(instance.executorsUtil));
+
+			EventCommonDataUtil.setBaseData(appContext);
+
+			AccountManagerImpl
+				.init(AccountManagerLocal.getInstance(appContext), eventLogger, AuthRepository.getInstance(),
+					BlockchainSourceImpl.getInstance());
+
+			OrderRepository.init(BlockchainSourceImpl.getInstance(),
+				eventLogger,
+				OrderRemoteData.getInstance(instance.executorsUtil),
+				OrderLocalData.getInstance(appContext, instance.executorsUtil));
+
+			OfferRepository.init(OfferRemoteData.getInstance(instance.executorsUtil), OrderRepository.getInstance());
+
+			DeviceUtils.init(appContext);
+
+			eventLogger.send(KinSdkInitiated.create());
 		}
 	}
 
-	public static void start(@NonNull Context appContext, @NonNull String jwt, @NonNull KinEnvironment environment)
-		throws ClientException, BlockchainException {
-		if (isInstanceNull()) {
-			SignInData signInData = getJwtSignInData(jwt);
-			init(appContext, signInData, environment);
+	private static void loadDefaultsFromMetadata(Context context) throws ClientException {
```

Comment:
do we really want to "break" the API and move configuration to xml? what are the benefits? 
consider when federation nodes will be up, you might need to re-expose a way to set a different blockchain node.

Reply:
1. In process restart we should get it without the user to call any method, and we will want to create few build and we don't want this configuration to be on runtime.
2. Not sure if we will need to expose anything to the developer related to the nodes, from my point of view we should handle something like that internally, not related to the developer.

Reply:
1. That is not requisite for process restart, you can persist `Environment` object and reload it on process start (you already doing that aren't you?)
2. That should be the developer call, to which blockchain node is connecting to.

Reply:
1. no, we don't want to persist that.
2. right now is not the case, maybe lated we will have another api method for overriding it.